submariner/OVN-Interconnect.md

```diff
@@ -0,0 +1,231 @@
+# Submariner Enhancement for OVN Interconnect
+
+<!-- Add link to issue/epic if available -->
+
+## Summary
+
+OVN CNI is moving to a new network topology powered by OVN Interconnect Feature. This allows independent OVN deployments to be interconnected
+by OVN-managed GENEVE tunnels. Submariner does not work with these changes out of the box and this proposal explains the changes required to
+support it.
+
+## Proposal
+
+With OVN Interconnect we can have two types of deployment
+
+* Single Zone(Global Zone): A single-zone deployment will have only one OVN database and a set of master nodes programming it. This is similar
+  to the topology that we use now. But it will have a single `global` zone assigned to it with a transit switch which is not involved in any
+  packet forwarding. Though this setup is expected to work by default, the subnet that the transit switch used overlaps with the subnet
+  Submariner uses for the logical router. In this proposal, we plan to remove the submariner router and the associated switches and this will
+  not be a problem anymore. The below indicates node annotations in a single zone  OVN setup.
+
+```bash
+    annotations:
+      k8s.ovn.org/ovn-node-transit-switch-port-ips: '["169.254.0.1/16"]'
+      k8s.ovn.org/ovn-zone: global
+    name: cluster1-worker
+
+    annotations:
+      k8s.ovn.org/ovn-node-transit-switch-port-ips: '["169.254.0.1/16"]'
+      k8s.ovn.org/ovn-zone: global
+    name: cluster2-worker
+```
+
+* Multiple Zone: In a multiple-zone setup, we will have an OVN database and a master pod for each zone. Transit switches connect the
+  zones. The OVN-Kubernetes services ensure that the necessary routes are added for pod and service reachability across nodes in different
+  zones.
+
+```bash
+    annotations:
+      k8s.ovn.org/ovn-node-transit-switch-port-ips: '["169.254.0.3/16"]'
+      k8s.ovn.org/ovn-zone: global
+    name: cluster1-worker
+
+    annotations:
+      k8s.ovn.org/ovn-node-transit-switch-port-ips: '["169.254.0.5/16"]'
+      k8s.ovn.org/ovn-zone: az2
+    name: cluster2-worker
+```
+
+With the current architecture, Submariner adds routes only in the zone in which it is deployed. For example, if Submariner is deployed in
+zone 1 it programs OVN db in zone 1. So only pods in zone 1 nodes will be able to talk to other clusters. Pods in zone 2 or zone 3 will not
+be able to reach remote clusters connected via Submariner.
+
+### Route APIs
+
+As part of this proposal, we are planning to add two new CRDs in Submariner
```

Comment:
It explains what CR will do but not why. Why do we need CR, why couldn't we directly program OVN rules. Are there plans for OVN to provide its own generic CRs for other components also to use?

Reply:
Added the details, aim is to decouple OVN dataplane programming and move it OVN repo in future.